AutoGear.lua

```diff
@@ -4108,10 +4119,7 @@ function AutoGearTooltipHook(tooltip)
 		)
 	end
 end
-GameTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
-ShoppingTooltip1:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
-ShoppingTooltip2:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
-ItemRefTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
+TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Item, AutoGearTooltipHook)
```

Comment:
This needs to support Classic clients still.  It should first check for the existence of `TooltipDataProcessor` and `TooltipDataProcessor.AddTooltipPostCall`, and if those don't exist, use the previous hooks, like this:

```Lua
if TooltipDataProcessor and TooltipDataProcessor.AddTooltipPostCall then
	TooltipDataProcessor.AddTooltipPostCall(Enum.TooltipDataType.Item, AutoGearTooltipHook)
else
	GameTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
	ShoppingTooltip1:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
	ShoppingTooltip2:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
	ItemRefTooltip:HookScript("OnTooltipSetItem", AutoGearTooltipHook)
end
```

Reply:
I simply attempted to fix the code the best I could. Leaving the older HookScript calls threw errors in Dragonflight which is where I am focused on playing currently. Hopefully someone else can dig into the Wrath issues and tooltip issues here. It seems to be working for the most part for my needs in Dragonflight.  Unfortunately I am not really available to dig into these issues further. Hopefully this will help move the add-on further for the current API.

Reply:
Even with my example, it didn't work?  If that's the case, Dragonflight is seriously messed up.

Reply:
I did not code your example. The changes I made are working for my needs. Not to sound like a jerk, I offered my changes as help to move this forward, not as your coder for your add-on which you are not actively coding for. I provided my changes to help not to be someone to do your code changes for you. If this is the change you are curious about feel free to install Dragonflight and Wrath classic and test it out. I am quite busy now with the holidays in a few days and don't have time between my full-time job, my family and the holidays to do this coding when you yourself admitted you don't have time. It's beginning to feel like you are expecting me to be your code monkey so you don't have to be bothered with maintaining your add-on. 

I was happy to help because I was able to get this add-on working for myself and no one else had made any progress for quite awhile after the pre-patch release. I don't have the time nor interest in being the guy to do the grunt work when it is not even my add-on and the owner doesn't have time to chip in either. Sorry if this sounds rude, I just wanted to contribute but each time I offer my changes, I seem to get back, well that isn't what I want, how about you do this instead.

Reply:
Yes, what you've said is considered rude in a pull request comment on github generally, but I understand that you aren't familiar with the process of pull requests and code reviews, so I don't have a problem with what you've said.  I'll help you understand.

This is how a pull request (or if on github, a merge request) works.  You submit changes you'd like merged, then if approved, they are merged.  If not approved, there is feedback explaining why.  We, as programmers, should not become upset by the process.  We are all volunteering our time, and we may have any number of other more prevalent interests for which to use our time.

In this particular case, your fork is extremely close to working well.  However, it's obvious to me that the tooltips would not even work in Dragonflight, let alone Wrath Classic, because of the additional check on which I commented, so I can't merge the changes as they are.  Since the addon still works in its supported environment, I will continue awaiting the 2 small fixes necessary before the merge.  I'd like to help you understand why those 2 small changes are necessary, as well, which is why I spent my time reviewing and commenting.  Just as you are more interested in Dragonflight, I am more interested in Wrath Classic, and I don't feel a particular need to merge unfinished and/or untested changes into a code base which is currently working.

Please don't imply that I am doing something wrong by following the usual github pull request process.  If you don't like upstream, you are free to continue to use your own fork of the project without resolving any issues found in your pull request's code review.  As an author of the original project, I allow this, and the project's license reflects that.